Casks/d/dnclient.rb

```diff
@@ -0,0 +1,41 @@
+cask "dnclient" do
+  version "0.9.0,c7546109"
+  sha256 "c9f372af6741eaf9c53c1ab94e9829ddd1bdacfc8ed2fbae8024f76a1d638c31"
+
+  url "https://dl.defined.net/#{version.csv.second}/v#{version.csv.first}/macos/DNClient-Desktop.dmg"
+  name "DNClient"
+  desc "Peer-to-peer VPN client for managed nebula networks"
+  homepage "https://www.defined.net/"
+
+  livecheck do
+    url "https://api.defined.net/v1/downloads"
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+    strategy :json do |json, regex|
+      json.dig("data", "dnclient")&.filter_map do |version, downloads|
+        next if version == "latest"
+        next unless downloads.key?("macos-universal-desktop")
+        next unless version.match?(regex)
+
+        download_url = downloads["macos-universal-desktop"]
+        git_sha = download_url[%r{/([a-f0-9]{8})/}, 1]
+        next unless git_sha
+
+        "#{version[regex, 1]},#{git_sha}"
+      end
+    end
+  end
```

Comment:
Thank you, @jasikpark! I have one more suggestion. Since the `latest`  release may be newer than the specific version tags, we should also extract the version from the download URL. How about the following:

```suggestion
  livecheck do
    url "https://api.defined.net/v1/downloads"
    regex(%r{/(\h+)/v?(\d+(?:\.\d+)+)/macos/DNClient-Desktop\.dmg}i)
    strategy :json do |json, regex|
      json.dig("data", "dnclient")&.map do |_, release|
        match = release["macos-universal-desktop"]&.match(regex)
        next if match.blank?

        "#{match[2]},#{match[1]}"
      end
    end
  end
```